### PR TITLE
allow converting decimal and integer ground terms to float64

### PR DESCRIPTION
**Pull request by matzemathics**

### CODE REVIEW

File: nemo/src/types.rs [hunk 1]
```diff
@@ -143,6 +143,15 @@ impl LogicalTypeEnum {
             },
             Self::Float64 => match gt {
                 Term::NumericLiteral(NumericLiteral::Double(d)) => DataValueT::Double(d),
+                Term::NumericLiteral(NumericLiteral::Decimal(a, b)) => {
+                    let shift = f64::log10(b as f64).floor() + 1.;
+                    DataValueT::Double(Double::from_number(
+                        a as f64 + b as f64 * 10f64.powf(-shift),
+                    ))
+                }
```
Comment (mmarx) [inline nemo/src/types.rs:151]: Please just use `format!("{a}.{b}").parse::<f64>()`, which will make sure that rounding happens correctly (and only once).

**Review by mmarx — CHANGES_REQUESTED** [review]

**Comment by monsterkrampe** [comment]
Is this supposed to be fixing #244 ?

File: nemo/src/types.rs [hunk 2]
```diff
@@ -143,6 +143,15 @@ impl LogicalTypeEnum {
...
+                Term::NumericLiteral(NumericLiteral::Integer(a)) => {
+                    DataValueT::Double(Double::from_number(a as f64))
```
Comment (monsterkrampe) [inline nemo/src/types.rs:153]: Similar handling should probably also introduced for the `RdfLiterals` that are `XSD_INTEGER` or `XSD_DECIMAL`, respectively (further below).

File: nemo/src/types.rs [hunk 3]
```diff
@@ -143,11 +143,17 @@ impl LogicalTypeEnum {
...
+                }
                 Term::RdfLiteral(RdfLiteral::DatatypeValue {
                     ref value,
                     ref datatype,
                 }) => match datatype.as_str() {
-                    XSD_DOUBLE => DataValueT::Double(
+                    XSD_DOUBLE | XSD_DECIMAL | XSD_INTEGER => DataValueT::Double(
```
Comment (mmarx) [inline nemo/src/types.rs:156]: Do we really want to implicitly convert integers to doubles, possibly losing precision?

Comment (matzemathics) [reply nemo/src/types.rs:156]: I have no real opinion about this. Generally I am against implicit conversions, but in this case we are talking about constants somebody types into a rules file. I could add a check that the parsed integer is smaller than 2**53 or disallow integer terms in double columns entirely, but then again, if the column type was defined to be double the user willingly allows loosing precision.

Comment (matzemathics) [reply nemo/src/types.rs:156]: Also there would be the possibility to allow 2 but disallow "2"^^xsd:integer. Not sure.

Comment (mmarx) [reply nemo/src/types.rs:156]: > Also there would be the possibility to allow 2 but disallow "2"^^xsd:integer. Not sure.

yeah, `2` should be accepted, but if somebody bothers to write `"2"^^xsd:integer`, I would expect that they care about XSD semantics, and those clearly state that `xsd:integer` and `xsd:double` have distinct value spaces.

Comment (mmarx) [reply nemo/src/types.rs:156]: Not sure how easy that would be to realise without major changes, though.

Comment (matzemathics) [reply nemo/src/types.rs:156]: On second thought, it feels that if someone tries to put a "2"^^xsd:integer into a float column, they probably made a mistake. So I will remove that cast for now.

**Review by mmarx — APPROVED** [review]

File: nemo/src/types.rs (hunk 3 above)
Comment (monsterkrampe) [reply nemo/src/types.rs]: Agreed :)

**Review by monsterkrampe — APPROVED** [review]

**Pull request merged**